easybuild/toolchains/fft/intelfftw.py

```diff
@@ -101,6 +101,7 @@ def _set_fftw_variables(self):
             # The cluster interface libs (libfftw3x_cdft*) can be omitted if the toolchain does not provide MPI-FFTW
             # interfaces.
             check_fftw_libs = [l for l in check_fftw_libs if l not in [interface_lib, cluster_interface_lib]]
+            fftw_libs = [l for l in fftw_libs if l not in [interface_lib, cluster_interface_lib]]
```

Comment:
@bartoldeman We're doing the exact same thing to both `check_fftw_libs` and `fftw_libs`, so let's refactor things a bit and pull down the definition of `check_fftw_libs` to right above where it is used:

```python
fftw_lib_exists = lambda x: any([os.path.exists(os.path.join(d, "lib%s.a" % x)) for d in fft_lib_dirs])
if not fftw_lib_exists(interface_lib) and LooseVersion(imklver) >= LooseVersion("10.2"):
    # interface libs can be optional:
    # ...
    fftw_libs = [l for l in fftw_libs if l not in [interface_lib, cluster_interface_lib]]

# filter out libraries from list of FFTW libraries to check for if they are not provided by Intel MKL
check_fftw_libs = [lib for lib in fftw_libs if lib not in ['dl', 'gfortran']]

if all([fftw_lib_exists(lib) for lib in check_fftw_libs]):
    self.FFT_LIB = fftw_libs
else:
    ...
```